### PR TITLE
Disable training when limit_train_batches=0

### PR DESCRIPTION
**Pull request by rohitgr7**
## What does this PR do?

when `limit_train_batches=0`, it still train the model for 1 batch. For now I have disable training completely in such a case, but it can be changed to disablevim epochs instead.

## Before submitting
- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? 
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review
 - [ ] Is this pull request ready for review? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.    
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by s-rog — APPROVED** [review]
Ran into this issue recently as well!

**Review by justusschock — APPROVED** [review]

**Review by SkafteNicki — APPROVED** [review]

File: tests/trainer/test_trainer.py [hunk 1]
```diff
@@ -748,6 +748,69 @@ def test_test_checkpoint_path(tmpdir, ckpt_path, save_top_k):
             assert trainer.tested_ckpt_path == ckpt_path
 
 
+def test_disabled_training(tmpdir):
+    """Verify that `limit_train_batches=0` disables the training loop unless `fast_dev_run=True`."""
+
+    class CurrentModel(EvalModelTemplate):
```
Comment (tchaton) [inline tests/trainer/test_trainer.py:754]: Can you use BoringModel ?

**Review by tchaton — CHANGES_REQUESTED** [review]
Great PR ! Would you mind using BoringModel ?

File: tests/trainer/test_trainer.py (hunk 1 above)
Comment (rohitgr7) [reply tests/trainer/test_trainer.py]: done.

**Review by tchaton — APPROVED** [review]
Great catch !

**Pull request merged**